src/Form/Type/ChangePasswordType.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Form\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\PasswordType;
+use Symfony\Component\Form\Extension\Core\Type\RepeatedType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Security\Core\Validator\Constraints\UserPassword;
+use Symfony\Component\Validator\Constraints\Length;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+/**
+ * Defines the custom form field type used to change user's password.
+ *
+ * @author Romain Monteil <monteil.romain@gmail.com>
+ */
+class ChangePasswordType extends AbstractType
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder
+            ->add('currentPassword', PasswordType::class, [
+                'constraints' => [
+                    new NotBlank(),
+                    new UserPassword(),
+                ],
+                'label' => 'label.current_password',
+                'attr' => [
+                    'autocomplete' => 'off',
+                ],
+            ])
+            ->add('newPassword', RepeatedType::class, [
+                'type' => PasswordType::class,
+                'constraints' => [
+                    new NotBlank(),
+                    new Length([
+                        'min' => 5,
+                    ]),
```

Comment:
I also suggest adding a `callback()` constraint and check by `BasePasswordEncoder::isPasswordTooLong($password)` otherwise we could get an uncatched error `BadCredentialsException` [here](https://github.com/symfony/demo/pull/834/files#diff-2e6355de36175249110ad2b48562407aR77).

Reply:
That's not something that would happen in a real-life scenario unless really trying to make it fail. No sensible user would try setting a 4096+ chars long password. So I would not showcase it here or instead arbitrary set a reasonable `max` value in `Length` constraint above.

Reply:
It's [72 chars for `BCryptPasswordEncoder`](https://github.com/symfony/symfony/blob/ce73293532f6df48b2cac2aad8d040817429cea3/src/Symfony/Component/Security/Core/Encoder/BCryptPasswordEncoder.php#L22) which is the current configured encoder :P but you're right, setting it (`BCryptPasswordEncoder::MAX_PASSWORD_LENGTH`) in `max` `Length` constraint is simple.